### PR TITLE
[SEDONA-480] Implement ST_S2ToGeom

### PR DESCRIPTION
**Pull request by furqaankhan**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-480. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?

- Expose the ST_S2ToGeom to public APIs.

## How was this patch tested?

- Passed new and old tests.

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

File: common/src/test/java/org/apache/sedona/common/FunctionsTest.java [hunk 1]
```diff
@@ -561,6 +561,27 @@ public void getGoogleS2CellIDsAllSameLevel() {
         assertEquals(expects, levels);
     }
 
+    @Test
+    public void testS2ToGeom() throws ParseException {
+        Geometry target = GEOMETRY_FACTORY.createPolygon(
```
Comment (jiayuasu) [inline common/src/test/java/org/apache/sedona/common/FunctionsTest.java:566]: Can you clean up all these comments and println?

File: snowflake-tester/src/test/java/org/apache/sedona/snowflake/snowsql/TestFunctions.java [hunk 2]
```diff
@@ -752,6 +752,7 @@ public void test_ST_S2CellIDs() {
...
         );
     }
+
```
Comment (jiayuasu) [inline snowflake-tester/src/test/java/org/apache/sedona/snowflake/snowsql/TestFunctions.java:755]: Please remove this change

File: spark/common/src/test/scala/org/apache/sedona/sql/functions/STS2CellIDs.scala [hunk 3]
```diff
@@ -100,4 +100,17 @@ class STS2CellIDs extends TestBaseScala with Matchers with GeometrySample with G
       )
     }
   }
+
+  describe("should pass ST_S2ToGeom") {
+    it("should intersect with the target geometry") {
```
Comment (jiayuasu) [inline spark/common/src/test/scala/org/apache/sedona/sql/functions/STS2CellIDs.scala:105]: Please add one more test to show that the result of ST_S2ToGeom can be used in ST_Collect to get a `MULTIPOLYGON`

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/test/java/org/apache/sedona/common/FunctionsTest.java (hunk 1 above)
Comment (furqaankhan) [reply common/src/test/java/org/apache/sedona/common/FunctionsTest.java:566]: My bad, I forgot about them.

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**